html/semantics/interactive-elements/the-dialog-element/backdrop-receives-element-events.html

```diff
@@ -0,0 +1,46 @@
+<!DOCTYPE html>
+<html>
+<title>Test that ::backdrop receives click events for the associated element</title>
+<link rel="author" title="Tim Nguyen" href="https://github.com/nt1m">
+<body>
+<style>
+/* ::backdrop takes up whole screen, actual <dialog> is hidden */
+dialog {
+    visibility: hidden;
+    pointer-events: none;
+}
+
+dialog::backdrop {
+    background-color: red;
+    top: 0;
+    bottom: 0;
+    left: 0;
+    right: 0;
+}
+
+dialog.clicked::backdrop {
+    background-color: green;
+}
+</style>
+<dialog></dialog>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/resources/testdriver.js"></script>
+<script src="/resources/testdriver-actions.js"></script>
+<script src="/resources/testdriver-vendor.js"></script>
+<script>
+const dialog = document.querySelector("dialog");
+dialog.showModal();
+async_test(function() {
+    dialog.addEventListener("click", () => {
+        dialog.className = "clicked";
+        this.done();
+    });
+    new test_driver.Actions()
```

Comment:
This returns a promise which could be rejected. Can you either handle that and fail the test (using `this.unreached_func('description')` or similar, or alternatively convert this to `promise_test` and turn the event into a promise instead? (Yes, tests that mix events and promises are pretty annoying.)

Or, as an option to simplify, you could use `setup({ single_test: true })`, that way a rejected promise will fail the single test without any glue code.

Reply:
Used the single_test setup, thanks!